tools/java/org/unicode/cldr/api/CldrDataSupplier.java

```diff
@@ -0,0 +1,241 @@
+ * order" and so any merging of values would have to track and rewrite sort indices carefully. It
+ * is recommended that if multiple {@code CldrData} instances are to be processed, users ensure
+ * that no path prefixes be shared between them. See also {@link CldrPath#getSortIndex()}.
+ *
+ * <p>Note that because the distinguishing paths associated with a {@link CldrValue} are unique per
+ * visitation, the special "version" path/value must be omitted (e.g. "//ldml/version") since it
+ * would otherwise appear multiple times. This should be fine, since the version is always available
+ * via {@link #getCldrVersionString()} and this mechanism is scheduled for deprecation anyway.
+ */
+public abstract class CldrDataSupplier {
+    /** Returns the current CLDR version string (e.g. {@code "36"}). */
+    public static String getCldrVersionString() {
+        return CLDRFile.GEN_VERSION;
+    }
+
+    /** Options for controlling how locale-based LDML data is processed. */
+    public enum CldrResolution {
+        /**
+         * Locale-based CLDR data should include resolved values from other "parent" locales
+         * according to the CLDR specification.
+         */
+        RESOLVED,
+
+        /**
+         * Locale-based CLDR data should only include values specified directly in the specified
+         * locale.
+         */
+        UNRESOLVED;
+    }
+
+    /**
+     * Returns a supplier for CLDR data in the specified CLDR project root directory. This must be
+     * a directory which contains the standard CLDR {@code "common"} directory file hierarchy.
+     *
+     * @param cldrRootDir the root directory of a CLDR project containing the data to be read.
+     * @return a supplier for CLDR data in the given path.
+     */
+    public static CldrDataSupplier forCldrFilesIn(Path cldrRootDir) {
+        return new FileBasedDataSupplier(
+            createCldrDirectoryMap(cldrRootDir), CldrDraftStatus.UNCONFIRMED);
+    }
+
+    /**
+     * Returns an unresolved CLDR data instance of a single XML file. This is typically only used
+     * for accessing additional CLDR data outside the CLDR project directories.
+     *
+     * @param type the expected CLDR type of the data in the XML file.
+     * @param xmlFile the CLDR XML file.
+     * @param draftStatus the desired status for filtering paths/values.
+     * @return a data instance for the paths/values in the specified XML file.
+     */
+    public static CldrData forCldrFile(CldrDataType type, Path xmlFile, CldrDraftStatus draftStatus) {
+        return new XmlDataSource(type, ImmutableSet.of(xmlFile), draftStatus);
+    }
+
+    private static Multimap<CldrDataType, Path> createCldrDirectoryMap(Path cldrRootDir) {
+        LinkedHashMultimap<CldrDataType, Path> multimap = LinkedHashMultimap.create();
+        for (CldrDataType type : CldrDataType.values()) {
+            type.getSourceDirectories()
+                .map(cldrRootDir.resolve("common")::resolve)
```

Comment:
`common` should not be codified here. Perhaps there should be a setter for the array of subdirs… perhaps by default `['common']` but maximally `['common','seed','exemplars','keyboards']`

yes, I know ldml2icu doesn't work this way. but it's broken. c.f. `SimpleFactory.make(File sourceDirectory[], …)` where an array of input directories is used.

So, this function shouldn't be static, with no other config data coming in but the path.

Reply:
This is an implementation detail and works fine for now, so I'd propose raising a bug (especially since I just don't have the context you do on this sort of thing). I'd be happy to extend the API to better understand some of these use cases, but for the NewIcu2LdmlConverter, this seems to be fine for now (I might realise it isn't as I convert the last few classes though, and if so, I'll fix it).